### PR TITLE
Add generic flashing script.

### PR DESCRIPTION
**Pull request by jusa**
Script can be configured in adaptation sparse/boot/flash-config.sh.

### CODE REVIEW

**Comment by sledges** [comment]
shellcheck fixes: https://github.com/sledges/droid-hal-configs/commit/35483b9ede6e4b316b73de1ad263e9686f0e6155

**Comment by jusa** [comment]
> shellcheck fixes: [sledges@35483b9](https://github.com/sledges/droid-hal-configs/commit/35483b9ede6e4b316b73de1ad263e9686f0e6155)

Thanks! Added to PR

**Review by sledges — APPROVED** [review]

File: sparse/boot/flash.sh [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+fi
+
+if [ -z "$FLASHCMD_FLASH_BOOT" ]; then
+    ### Detect fastboot flash commands
+
+    # By default use "flash" for boot partition
+    FLASHCMD_FLASH_BOOT="flash"
+
+    # If flash:raw command exists use that instead
+    if ${FASTBOOT_BIN_PATH}${FASTBOOT_BIN_NAME} help | grep -q flash:raw; then
```
Comment (sledges) [inline sparse/boot/flash.sh:285]: Even old fastboot has "flash:raw", but it outputs everything to stderr, so this grep is not 100% accurate:)
old fastboot:
stderr: `flash:raw boot <kernel> [ <ramdisk> [ <second> ] ] create bootimage and`
new fastboot:
stdout: `flash:raw PARTITION KERNEL [RAMDISK [SECOND]]`

Comment (sledges) [reply sparse/boot/flash.sh:285]: And that's why during this line the old `fastboot help` outputs all help screen to our device's flashing console log.

Comment (sledges) [reply sparse/boot/flash.sh:285]:
```diff
-    if ${FASTBOOT_BIN_PATH}${FASTBOOT_BIN_NAME} help | grep -q flash:raw; then
+    if ${FASTBOOT_BIN_PATH}${FASTBOOT_BIN_NAME} help 2>&1 | grep -q "flash:raw PARTITION"; then
```

**Pull request merged**